src/main/java/com/stripe/net/OAuth.java

```diff
@@ -0,0 +1,82 @@
+import com.stripe.net.APIResource;
+import com.stripe.net.LiveStripeResponseGetter;
+import com.stripe.net.RequestOptions;
+import com.stripe.net.StripeResponseGetter;
+
+public final class OAuth {
+	private static StripeResponseGetter stripeResponseGetter = new LiveStripeResponseGetter();
+
+	public static void setStripeResponseGetter(StripeResponseGetter srg) {
+		OAuth.stripeResponseGetter = srg;
+	}
+
+	public static String authorizeURL(Map<String, Object> params, RequestOptions options) throws AuthenticationException, InvalidRequestException {
+		String base = Stripe.getConnectBase();
+
+		params.put("client_id", getClientId(params, options));
+		if (params.get("response_type") == null) {
+			params.put("response_type", "code");
+		}
+		String query;
+		try {
+			query = LiveStripeResponseGetter.createQuery(params);
+		} catch (UnsupportedEncodingException e) {
+			throw new InvalidRequestException("Unable to encode parameters to "
+					+ APIResource.CHARSET
+					+ ". Please contact support@stripe.com for assistance.",
+					null, null, 0, e);
+		}
+
+		String url = base + "/oauth/authorize?" + query;
+		return url;
+	}
+
+	public static TokenResponse token(Map<String, Object> params, RequestOptions options) throws AuthenticationException, InvalidRequestException, APIConnectionException, APIException, OAuthException {
+		String url = Stripe.getConnectBase() + "/oauth/token";
+		return OAuth.stripeResponseGetter.oAuthRequest(APIResource.RequestMethod.POST, url, params, TokenResponse.class,
+				APIResource.RequestType.NORMAL, options);
+	}
+
+	public static DeauthorizedAccount deauthorize(Map<String, Object> params, RequestOptions options) throws AuthenticationException, InvalidRequestException, APIConnectionException, APIException, OAuthException {
+		String url = Stripe.getConnectBase() + "/oauth/deauthorize";
+		params.put("client_id", getClientId(params, options));
+		return OAuth.stripeResponseGetter.oAuthRequest(APIResource.RequestMethod.POST, url, params, DeauthorizedAccount.class,
+				APIResource.RequestType.NORMAL, options);
+	}
+
+	private static String getClientId(Map<String, Object> params, RequestOptions options) throws AuthenticationException {
+		String clientId = Stripe.clientId;
+		if ((options != null) && (options.getClientId() != null)) {
+			clientId = options.getClientId();
+		}
+		if ((params != null) && (params.get("client_id") != null)) {
+			clientId = (String) params.get("client_id");
+		}
+
+		if (clientId == null) {
+			throw new AuthenticationException(
+					"No client_id provided. (HINT: set client_id key using 'Stripe.client_id = <CLIENT-ID>'. "
+							+ "You can find your client_ids in your Stripe dashboard at "
+							+ "https://dashboard.stripe.com/account/applications/settings, "
+							+ "after registering your account as a platform. See "
+							+ "https://stripe.com/docs/connect/standard-accounts for details, "
+							+ "or email support@stripe.com if you have any questions.",
```

Comment:
I think we'll have to change this error message right?

Reply:
It's the same error message as in the other language implementations (e.g. in [stripe-ruby](https://github.com/stripe/stripe-ruby/blob/a645a78cd028554165e44760d531f946db0ea1cb/lib/stripe/oauth.rb#L18-L24)). What do you think needs to be changed?

Reply:
@ob-stripe Sorry, I should have been more specific. The config is called `Stripe.clientId` in Java (not `Stripe.client_id`).

Reply:
🤦‍♂️  Of course, thanks!